src/core/render.ts

```diff
@@ -1,57 +1,232 @@
 /// <reference path="../../defs/es6-promise.d.ts" />
 
 import {Scene} from './scene';
+import {Item} from './items/item';
+import {Source} from './source/source';
 import {App as iApp} from '../internal/app';
 import {Render as iRender} from '../internal/render';
 import { exec } from '../internal/internal';
 
 export class Render {
+  private static _canvasCounter: number = 0;
+  private _index: number;
+  private _fps: number;
+  private _canvas: HTMLCanvasElement;
+  private _rendering: boolean;
+  private static _canvasPool: Render[] = [];
 
+  constructor(props?) {
+    this._index = props.canvasIndex;
+    this._fps = props.fps;
+    this._canvas = props.canvas;
+    this._rendering = props.rendering;
+  }
 
-  // Assign a canvas and create necessary webgl context
-  static setTargetCanvas(canvas: HTMLCanvasElement,
-    fps: number = 30): Promise<any> {
+  static getPreparedCanvases() {
+    return new Promise(resolve => {
+      resolve(Render._canvasPool)
+    })
+  }
+
+  /**
+   * param: (canvas: HTMLCanvasElement, fps?: number)
```

Comment:
@SML-MeSo we did not document the 3rd parameter. Is this intentional
```suggestion
   * param: (canvas: HTMLCanvasElement, fps?: number, includeToPool?: boolean)
```